risc0/zkvm/src/exec/monitor.rs

```diff
@@ -83,6 +88,7 @@ impl MemoryMonitor {
         let pages = vec![None; num_pages];
         Self {
             image,
+            num_pages,
```

Comment:
I think you can grab this from the `image.info`?

Reply:
Ah ok, looking at more lines in the diff and I now see it is coming from `image.info`. We're just caching this value on the `MemoryMonitor` for easy access. OK, seems fine to me.